cookiecutter/exceptions.py

```diff
@@ -113,3 +113,26 @@ class RepositoryNotFound(CookiecutterException):
     """
     Raised when the specified cookiecutter repository doesn't exist.
     """
+
+
+class RepositoryCloneFailed(CookiecutterException):
+    """Raised when a cookiecutter template can't be cloned."""
+
+    BRANCH_ERRORS = [
+        'error: pathspec',
+        'unknown revision',
+    ]
+
+    def __init__(self, exc, repo_url, checkout):
+        if 'not found' in exc.output.lower():
+            super(CookiecutterException, self).__init__(
```

Comment:
So this will use ``Exception`` rather than ``CookiecutterException``. I think we should use the latter instead.